# IlluminaBacSeq
A simple pipeline for de novo genome assembly of clinical bacterial isolates using short-read sequencing data  

The pipeline is designed to process short-read sequencing data, perform trimming, de novo assembly, quality assessment and provide a summary of the assembly metrics. This pipeline was specifically designed to work with our own dataset, however the trimming and assembly parameters can be changed by altering the respective scripts to suit the user's needs.

To read more, click on the following link:
https://docs.google.com/document/d/e/2PACX-1vSC5OIyYUQUYe7Bukhd_6UhQkTZzsMkMAzKER3qv_b8nSxHSMhjWNIwetp6m_mpt-kmblYGlAOG0lJ6/pub

# References
1.	Bolger, A. M., Lohse, M., & Usadel, B. (2014). Trimmomatic: a flexible trimmer for Illumina sequence data. Bioinformatics, 30(15), 2114–2120. https://doi.org/10.1093/BIOINFORMATICS/BTU170

2.	Gurevich, A., Saveliev, V., Vyahhi, N., & Tesler, G. (2013). QUAST: quality assessment tool for genome assemblies. Bioinformatics, 29(8), 1072–1075. https://doi.org/10.1093/BIOINFORMATICS/BTT086

3.	Wick, R. R., Judd, L. M., Gorrie, C. L., & Holt, K. E. (2017). Unicycler: Resolving bacterial genome assemblies from short and long sequencing reads. PLOS Computational Biology, 13(6), e1005595. https://doi.org/10.1371/JOURNAL.PCBI.1005595

Organization: Ashoka University


